core/src/main/java/nl/inl/util/LuceneUtil.java

```diff
@@ -327,15 +327,13 @@ public static void getFrequenciesFromTermVector(IndexReader reader, int doc,
                                 for (BytesRef term = termsEnum.term(); term != null; term = termsEnum.next()) {
                                     if (maxResults < 0 || results.size() < maxResults) {
                                             String termText = term.utf8ToString();
-                                            String optDesensitized = termText;
-                                            if (!sensitive)
-                                                    optDesensitized = StringUtil.stripAccents(termText).toLowerCase();
-                                            if (!allTerms && !optDesensitized.substring(0, prefix.length()).equalsIgnoreCase(prefix)) {
+                                            boolean startsWithPrefix = sensitive ? termText.startsWith(prefix): StringUtil.insensitiveStartsWith(termText, prefix);
+                                            if (!allTerms && !startsWithPrefix) {
                                                     // Doesn't match prefix or different field; no more matches
                                                     break;
                                             }
                                             // Match, add term
-                                            results.add(termText);
+                                            if (!results.contains(termText)) results.add(termText);
```

Comment:
We could use a HashSet<> instead, the order is not relevant anyway (because terms are collected from multiple LeafReaders).

Reply:
If you want sorting, a List can be sorted, a set cannot be.

I think it would be good to call Collections.sort before returning the list

Reply:
Sure. Personally I would collect things in a set, then convert to a list and sort it. The n^2 complexity of those linear searches just makes me itch a little. :-)

Reply:
you expect it to be cheaper that way?

Reply:
ok, I''l make it a TreeSet then